quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/LoggingPropertyMappers.java

```diff
@@ -192,6 +199,15 @@ private static void setCategoryLevel(String category, String level) {
 
     record CategoryLevel(String category, String levelName) {}
 
+    private static void validateLogParameters(Option<LoggingOptions.Level> option, String level) {
+        try {
+            toLevel(level);
+        } catch (IllegalArgumentException iae) {
+            String expectedValues = Stream.of(option.getExpectedValues()).map(Object::toString).collect(Collectors.joining(", ")).replace("[","").replace("]","");
```

Comment:
One final thought - we could create PropertyMapper.validateExpectedValues could be refactored so that a common method supports both the default validation and this one:

```
void validateExpectedValues(ConfigValue configValue, String v, Predicate<String> test)
```

Where the default test is "!expectedValues.isEmpty() && !expectedValues.contains(v) && getOption().isStrictExpectedValues()"

Reply:
To elaborate, on PropertyMappers this would look like:

```
    void validateExpectedValues(ConfigValue configValue, String v) {
        if (getOption().isStrictExpectedValues()) {
            validateExpectedValues(configValue, v, (expectedValues) -> !expectedValues.isEmpty() && expectedValues.contains(v));
        }
    }

    void validateExpectedValues(ConfigValue configValue, String v, Function<Collection<String>, Boolean> test) {
        List<String> expectedValues = getExpectedValues();
        if (test.apply(expectedValues)) {
            throw new PropertyException(
                    String.format("Invalid value for option %s: %s.%s", getOptionAndSourceMessage(configValue), v,
                            ShortErrorMessageHandler.getExpectedValuesMessage(expectedValues)));
        }
    }
```

Then you would call it on the logging proeprty mappers:

```
                fromOption(LoggingOptions.LOG_SYSLOG_LEVEL)
                        .isEnabled(LoggingPropertyMappers::isSyslogEnabled, SYSLOG_ENABLED_MSG)
                        .validator(ignored -> {})
                        .addValidator((mapper, configValue) -> mapper.validateExpectedValues(configValue,
                                configValue.getValue(),
                                (expectedValues) -> expectedValues.contains(configValue.getValue().toLowerCase())))
```

That will make the exception handling logic the same.

The other possiblities here are:
- declaratively mark the option as case-insensitive, so that the validateExpectedValues does the lower-casing automatically (similar to the isStrictExpectedValues check) and we don't have to directly mess with the validators on the logging property mappers.
- add another builder validator method so we don't need to call both validator and addValidator

Reply:
Hi @shawkins , I tried to incorporate the changes (Updated the PR) . The build is successful but i still get `ERROR: The LogManager accessed before the "java.util.logging.manager" system property was set to "org.jboss.logmanager.LogManager". Results may be unexpected.`
<img width="1516" alt="Screenshot 2024-11-13 at 7 43 57 PM" src="https://github.com/user-attachments/assets/f086ec3f-ae46-4bcd-817e-f5cc7da162c1">

Reply:
That error likely indicates an uncaught exception that is trigger some logging at another point in the code. You'll have to debug to see what is happening.